src/ReachSets/ContinuousPost/BFFPS19/BFFPS19.jl

```diff
@@ -0,0 +1,291 @@
+            end
+        end
+    end
+
+    nothing
+end
+
+"""
+    BFFPS19 <: ContinuousPost
+
+Implementation of the reachability algorithm for purely continuous linear
+time-invariant systems using block decompositons by S. Bogomolov, M. Forets,
+G. Frehse, A. Podelski, C. Schilling and F. Viry [1].
+
+### Fields
+
+- `options` -- an `Options` structure that holds the algorithm-specific options
+
+### Notes
+
+The following options are available:
+
+```julia
+$(print_option_spec(options_BFFPS19()))
+```
+
+### Algorithm
+
+We refer to [1] for technical details.
+
+[1] [Reach Set Approximation through Decomposition with Low-dimensional Sets
+and High-dimensional Matrices](https://dl.acm.org/citation.cfm?id=3178128).
+S. Bogomolov, M. Forets, G. Frehse, A. Podelski, C. Schilling, F. Viry.
+HSCC '18 Proceedings of the 21st International Conference on Hybrid Systems:
+Computation and Control (part of CPS Week).
+"""
+struct BFFPS19 <: ContinuousPost
+    options::TwoLayerOptions
+
+    function BFFPS19(𝑂::Options)
+        normalized_𝑂 = validate_and_wrap_options(𝑂, options_BFFPS19();
+            validation=validation_BFFPS19,
+            normalization=normalization_BFFPS19!)
+        return new(normalized_𝑂)
+    end
+end
+
+# convenience constructor from pairs of symbols
+BFFPS19(𝑂::Pair{Symbol,<:Any}...) = BFFPS19(Options(Dict{Symbol,Any}(𝑂)))
+
+# default options
+BFFPS19() = BFFPS19(Options())
+
+include("reach.jl")
+include("reach_blocks.jl")
+
+init(𝒫::BFFPS19, 𝑆::AbstractSystem, 𝑂::Options) = init!(𝒫, 𝑆, copy(𝑂))
+
+function init!(𝒫::BFFPS19, 𝑆::AbstractSystem, 𝑂::Options)
+    # state dimension for (purely continuous or purely discrete systems)
+    𝑂copy = copy(𝑂)
+    𝑂copy[:n] = statedim(𝑆)
+
+    # solver-specific options (adds default values for unspecified options)
+    𝑂validated = validate_solver_options_and_add_default_values!(𝑂copy)
+
+    # :vars option; default: all variables
+    if haskey_specified(𝒫.options, :vars)
+        𝑂validated[:vars] = 𝒫.options[:vars]
+    else
+        𝑂validated[:vars] = 1:𝑂validated[:n]
+    end
+
+    # :partition option: use 1D blocks
+    if haskey_specified(𝒫.options, :partition)
+        𝑂validated[:partition] = 𝒫.options[:partition]
+    else
+        𝑂validated[:partition] = [[i] for i in 1:𝑂validated[:n]]
+    end
+
+    opD = 𝒫.options[:opD]
+    if opD isa DecomposedDiscretePost
+        HS = 𝒫.options[:HS]
+        constrained_dims = constrained_dimensions(HS)
+        out_vars = opD.options[:out_vars]
+        loc_id = 𝒫.options[:loc_id]
+        temp_vars = unique([out_vars; constrained_dims[loc_id]])
+        opD.options[:temp_vars] = temp_vars
+        guards_constraints = [guard(HS, trans) for trans in out_transitions(HS, loc_id)]
+        𝑂validated[:vars] = temp_vars
+        𝑂validated[:guards_proj] = [project(c, temp_vars) for c in guards_constraints]
+        𝑂validated[:blocks] = compute_blocks(𝑂validated[:vars], 𝑂validated[:partition])
+
+        info("- Total")
```

Comment:
Is there a reason why you moved this line here?

Reply:
Nope, just copied too many lines:)